docs/design/resources.md

```diff
@@ -0,0 +1,96 @@
+# Operator Resources
+
+The etcd cluster operator extends the Kubernetes API with two new
+resource definitions:
+
+* `EtcdCluster`, or the "cluster resource", defines an entire
+  operating cluster.
+* `EtcdPeer`, or the "peer resource", defines a single peer of the
+  etcd cluster. This principally concerns a Pod, but may also involve
+  attached storage and network identity.
+
+## The Peer Resource
+
+The peer resource represents a single peer in an etcd
+cluster. Encoding it's bootstrap instructions, persistence settings,
+and extra pod settings (e.g., overridden CPU limits). Its `status`
+field exposes information about the peer including its hostname,
+liveness status, the peer's current generation, and if it's currently
+the leader of the etcd cluster.
+
+When reconciling the etcd peer resource the operator will do the
+following:
+
+1. If storage is enabled and a correctly named persistent volume claim
+   (PVC) does not exist, create one with the settings provided on the
+   peer resource.
+
+   We never delete a persistent volume claim automatically, so this
+   behaviour ensures that the operator will resume stopped clusters if
+   the storage remains even if the peer resource was deleted.
+
+   If the storage options on the peer resource are different from an
+   existing PVC, the operator makes no attempt to 'correct' the
+   preexisting PVC.
+2. If a correctly named pod does not exist, an etcd pod will be
+   launched with the PVC from above mounted as a volume if storage is
+   enabled. The bootstrap configuration will always be provided to the
+   pod as configuration, as etcd will ignore bootstrap configuration
+   if a data directory already exists.
+3. If the pod is in a ready state, the operator will attempt to
+   contact it using the gRPC endpoint and retrieve information to
+   populate the status field. Otherwise, it will report the peer as
+   not-ready on the status field.
+
+## The Cluster Resource
+
+The cluster resource provides management of the entire cluster. It
+will ensure that the correct number of `EtcdPeer` resources exist,
+deploys a "headless" service to allow connectivity to the cluster, and
+keeps its status field updated with the information about the cluster.
+
+The operator treats the cluster itself as the canonical view of the
+current state of the world. In particular the etcd cluster's peer list
+is used to define what the expected members should be.
```

Comment:
Maybe follow on to explain what the role of the cluster resource is here, i.e. "the cluster's peer list is used to define what EtcdPeer resources should exist - if a user requests a 'scale down', the member is removed from the etcd membership API. This in turn will cause the EtcdPeer resource to be removed, as the peer will no longer be present in the membership API"?

Reply:
I was going to explain the scale/up down in a worked example below. Do you still think it's best to modify this section?

Reply:
Nope, np 😄